plasmapy/simulation/particle_integrators.py

```diff
@@ -36,6 +36,29 @@ def boris_push(x, v, B, E, q, m, dt):
     dt : `float`
         Timestep, in SI (second) units.
 
+    Examples
+    --------
+    >>> from plasmapy.simulation.particle_integrators import boris_push
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>>
+    >>> x_0 = np.array([0.0, 0.0, 0.0])
+    >>> v_0 = np.array([5.0, 0.0, 0.0])
+    >>> B = np.array([0.0, 0.0, -5.0])
+    >>> x = []
+    >>> y = []
+    >>>
+    >>> for _ in range(150):
+    >>>     boris_push(x = x_0, v = v_0, B=B, E=0, q=1.0, m=1.0, dt=0.01)
```

Comment:
So I just realized that the `boris_push` implementation we've got here only modifies the input `x, v` arrays. This is good for performance but is dreadful for quickly displaying how it works. I think adding an `inplace = False` keyword argument would be okay both for docs and for future developments here (for example I have a couple ideas on doing this through jax or some such optimizer...), if you'd be up for it!

And then, rather than having an embedded minimal simulation here, I think it'd be better to show how how to use the boris pusher:

1. how it returns the x, v arrays in the next time step (the essence of what it does) 
2. how it modifies the x, v arrays by default (show them before/after)
3. how, for B parallel to V, it does nothing;
4. what it does for B perpendicular to V;
5. how for nonzero E and zero B it reduces to the leapfrog algorithm.

Also, to easily do this from the IPython shell, try the `%doctest_mode` magic command: 

```
[ins] In [1]: %doctest_mode
Exception reporting mode: Plain
Doctest mode is: ON
>>> 2 + 2
4
```

Reply:
Ok I will work on that and prepare new version in accordance with your suggestions.